builtins/src/cost_modeling.rs

```diff
@@ -0,0 +1,162 @@
+//! Configurations for handling cost modeling of builtin programs.
+
+use {
+    crate::BUILTINS,
+    ahash::AHashMap,
+    lazy_static::lazy_static,
+    solana_feature_set::FeatureSet,
+    solana_pubkey::Pubkey,
+    solana_sdk_ids::{ed25519_program, secp256k1_program},
+};
+
+/// Configuration for cost modeling of a builtin program.
+#[derive(Debug)]
+pub struct CostModelingConfig {
+    /// The default cost of the builtin program.
+    /// If None, this builtin program is not cost modeled.
+    pub default_cost: Option<u64>,
```

Comment:
Nice, this solves ZK stuff. We could use `enum` instead of `Option` to make it even more distinguished.

Reply:
I refactored to use an enum! I think that's a smoother approach. Good tip.